Calvin Passmore

ECE 6600

Feb 27, 2023

# Homework 2

## Problem 1

How much is 50mW in dBm? How much is it in dBW

---

10 log10 (50 mW / 1 mW) = 16.98970 dBm

10 log10 (50 mW / 1 W) = -13.01 dBW

---
---

## Problem 2
For each carrier frequency given below, determine the received power (in dBm) in free space of  a signal whose transmit power is 4W if the receiver is at distance of 5 km from the transmitter?

What is the path loss in dB?

a. Cellular Band transmitter: 830 MHz

b. PCS Band transmitter: 1880 MHz

c. WLAN Band transmitter: 5.5 GHz

---

Pr_dBm = Pt_dBm -21.98 + 20 log<sub>10</sub>(&lambda;) - 20 log<sub>10</sub>(d)

Pt = 4W
d = 5000m
lambda = c/f

a. 

Pr_dBm = 4 -21.98 + 20 log<sub>10</sub>(c/830M) - 20 log<sub>10</sub>(5000) = -100.8045 dBm

b.

Pr_dBm = 4 -21.98 + 20 log<sub>10</sub>(c/1880M) - 20 log<sub>10</sub>(5000) = -107.9061 dBm

c.

Pr_dBm = 4 -21.98 + 20 log<sub>10</sub>(c/5.5G) - 20 log<sub>10</sub>(5000) = -117.2302 dBm

---
---

## Problem 3
Consider a mobile phone system cell using a frequency of 900 MHz providing coverage to a urban environment like the downtown of Salt Lake City. The height of the receiver is 2 meters and the cell site antenna is 34 meters tall. The mobile terminal receivers have a sensitivity of –90 dBm. Using the Okumura---Hata propagation model determine what power level the base station must operate at to have a cell coverage radius of 1.6 km.

---


In [8]:
from math import log10

fc = 900e6
hre = 2
hte = 34
d= 1.6
power_limit = -90 # dBm

Lp_dB = 69.55 + 26.16*log10(fc) - 13.82 * log10(hte) - 3.2 * (log10(11.75 * hre))**2 - 4.97 + (44.9 - 6.55 * log10(hte)) * log10(d)

Lp_dBm = Lp_dB + 30

# Needed power
Power = Lp_dBm + power_limit

print(Power)

218.75992027180678


so 219 dBm is needed

---
---

## Problem 4
What is the fading margin if the shadow fading component has a standard deviation  = 8dB and you want to design a system that has 95% coverage at the edge of the cell?

---

P(X > F) = 0.05

F &approx; 13.125

---
---

## Problem 5

If you apply that fading margin to the mobile phone system in Problem 3, at what power level must the base station operate?

---

13.125 + 219 = 232.125 dB

---
---

## Problem 6

When is a radio channel "narrowband" or "wideband"?

---

A channel is wideband when it's operational bandwidth exceeds the coherence bandwidth, and is narrow band when the operational bandwidth is less than the coherence bandwidth

---
---

## Problem 7

A wireless transmission scheme employs 16 QAM and a symbol is sent every millisecond. 
What is the data rate of the system in kbps?

---

16 QAM = 4 bits/symbol
1000 symbols / sec

= 4 kbps

---
---


## Problem 8

Consider a (48, 36) block error correcting code used in the downlink control channel of a cellular system.
This code can correct up to 5 bit errors in a block of 48 bits.

a. What is the code rate?

b. What is the maximum error burst size that can be tolerated and still have correct reception of the data?

c. An error burst is usually caused by time variation of the channel in cellular networks.
If the control channel operates at 48 kbps, what is the maximum fade duration (in seconds) that can be tolerated?

d. If the 48 bit blocks are buffered until 10 blocks are collected and then they are interleaved and transmitted, what is the maximum error burst size that can be tolerated and still have correct reception of the data?

e.  What is the delay incurred in buffering the data for interleaving and de-interleaving at a 48 kbps rate?

---

a. 3/4 code rate

b. A burst size of 4 bits in a single block could be corrected, or if the burst crossed two blocks evenly 8 bits could be corrected.

c. 83 us

d. 40 bit error burst size

e. The delay is 9 ms to wait for the rest of the 9 blocks to be buffered, for a total wait time of 10 ms.


In [6]:
# a (n,k) block
n = 48
k = 36
code_rate = k/n
print(f"Coding rate {code_rate}")

# b
bits_per_second = 48000
max_bit_errors = 4
seconds_per_bit = 1/bits_per_second
acceptable_fade = max_bit_errors * seconds_per_bit
print(f"Max fade duration {acceptable_fade} s")

# e
seconds_per_block = 48 * seconds_per_bit
blocks_per_buffer = 10
seconds_per_buffer = blocks_per_buffer * seconds_per_block
print(f"Total buffering time {seconds_per_buffer} s")
print(f"Delay from extra buffering {seconds_per_buffer - seconds_per_block} s")

Coding rate 0.75
Max fade duration 8.333333333333333e-05 s
Total buffering time 0.01 s
Delay from extra buffering 0.009000000000000001 s



---
---



## Problem 9

Consider a cellular system that uses frequency hopping and has 124 frequency channels of spectrum.
The hopping pattern for a user in a cell that has 31 frequency channels assigned to it is shown below.

a.  Is the hopping pattern random or cyclic?

b. Is this a slow or fast frequency hopping system? Why?

---

a. The hopping pattern appears to be random, seeing as it never repeats and doesn't follow an obvious pattern

b. Fast hopping switches frequencies either every bit or fractions of a bit, the frequencies shown in this problem switch every 2 bits, and are therefore slow hops.

---
---